backends/dpdk/control-plane/bfruntime_ext.cpp

```diff
@@ -213,7 +213,25 @@ const Util::JsonObject*
 BFRuntimeSchemaGenerator::genSchema() const {
     auto* json = new Util::JsonObject();
 
-    json->emplace("schema_version", cstring("1.0.0"));
+    if (isTDI) {
+        cstring progName =  options.file;
+        auto fileName = progName.findlast('/');
+        // Handle the case when input file is in the current working directory.
+        // fileName would be null in that case, hence progName should remain unchanged.
+        if (fileName)
+            progName = fileName;
+        auto fileext = progName.find(".");
+        progName = progName.replace(fileext, "");
+        progName = progName.trim("/\t\n\r");
+        json->emplace("program_name", progName);
+        json->emplace("build_date", cstring(options.getBuildDate()));
+        json->emplace("compile_command", cstring(options.getCompileCommand()));
+        json->emplace("compiler_version", cstring(options.compilerVersion));
+        json->emplace("schema_version", cstring("0.1"));
+        json->emplace("target", cstring("DPDK"));
+    } else {
+        json->emplace("schema_version", cstring("1.0.0"));
```

Comment:
this one as well

Reply:
Version strings can have more than one decimal point. So I think a string literal should be ok here.
Do you mean instead of string literal, we should use a const string type variable to hold the version string?

Reply:
Yes, a const string, in one place.